L2/hardfork_devnet_test.md

```diff
@@ -0,0 +1,95 @@
+    - Download genesis.json into your working directory and rename it:
+      ```bash
+      mv genesis.json qkc.json
+      ```
+    - Install compression tool if necessary:
+      ```bash
+      apt install zstd
+      ```
+    - Compress genesis file using the provided dictionary:
+      ```bash
+      zstd -D ./superchain/extra/dictionary qkc.json
+      ```    
+    - Move compressed file to correct location:
+      ```bash
+      mv qkc.json.zst ./superchain/extra/genesis/sepolia/
+      ```        
+  - Commit and push changes:
+    ```bash
+    git add .
+    git commit -m "Update genesis and config for L2Blob and SGT enabling"
+    git push origin devnet    
+    ```  
+## 3. Generating the Latest `superchain-configs.zip`
+We will generate the zip file on the deployment machine directly.
+  - Modify `op-geth` source:
+    - Navigate to the `op-geth` folder.
+    - Update repository URL in `sync-superchain.sh`
+      ```bash
+      - https://github.com/ethereum-optimism/superchain-registry.git
+      + https://github.com/QuarkChain/superchain-registry.git      
+      ```
+    - Update commit hash in `superchain-registry-commit.txt` to match the latest commit on the devnet branch.
+  - Run `sync-superchain.sh` to generate the new `superchain-configs.zip`
+  - Rebuild binaries: 
+    - For op-geth: 
+      ```bash
+      make geth
+      ```
+    - For op-node: 
+      - Copy `superchain-configs.zip` to the appropriate library folder.
+      - Temporarily comment out the CheckL1ChainID validation [here](https://github.com/QuarkChain/optimism/blob/06a9487cb7f3b9398de0b9ba27896e7a4ef9d1c0/op-node/rollup/types.go#L195)
+      - Compile op-node:
+        ```bash
+        cd ../optimism
+        just op-node/op-node
+        ```
+> ⚠️ Important Notes: 
+>   - For mainnet deployment, we should dump and compare the results from LoadOPStackGenesis and LoadOPStackRollupConfig to the corresponding JSON outputs. This comparison ensures that no fields have been unintentionally modified, significantly reducing testing effort and helping us quickly validate the integrity of configurations.
+>   - For mainnet hardfork release, we should first commit the updated `superchain-configs.zip` and `superchain-registry-commit.txt` to the op-geth repository. Afterward, we’ll build the final release binary and validate it thoroughly in our staging environment to ensure everything operates as expected. Once testing is complete and confirmed successful, we can then officially release the final version and deploy to production.        
+## 4. Restart op-node && op-geth
+  - Restart op-node
+    - Remove `--rollup.config` flag if currently used
+    - Add the `--network=qkc-sepolia` flag to specify the updated configuration.
+  - Restart op-geth
+    - Add the `--op-network=qkc-sepolia` flag to specify the updated configuration.
+
+## 5. Final Testing for L2Blob and SGT
+After completing all previous steps, conduct your tests again:
```

Comment:
Do we also have to wait for it to pass the timestamp?

Reply:
Yes